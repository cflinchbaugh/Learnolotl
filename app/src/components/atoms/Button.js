import React, { Component } from 'react';
import styled from 'styled-components';

const StyleWrapper = styled.div`
    .button {
        background-color: ${props => props.theme.secondary};
        color: ${props => props.theme.secondaryColor};
        transition: all 0.35s; 
        display: inline-block;
        border-radius: 15px;
        padding: 8px 20px;
        box-shadow: 1px 8px 19px 3px rgba(61,55,61,0.25);
        user-select: none;

        &:hover {
            cursor: pointer;
            background-color: ${props => props.theme.secondaryLight};
            color: ${props => props.theme.secondaryLightColor}
            box-shadow: 1px 8px 19px 3px rgba(61,55,61,0.48);
        }
    }
`

function Button (props) {
    return (
        <StyleWrapper>
            <div className="button" 
                onClick={ props.onClickFunction } 
                onMouseOver={ props.onMouseOverFunction }
            >
                { props.label }
            </div>
        </StyleWrapper>
    )
}

export default Button