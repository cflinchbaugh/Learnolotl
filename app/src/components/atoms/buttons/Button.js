import React from 'react';
import styled from 'styled-components';

const StyleWrapper = styled.div`
    .button {
        background-color: ${props => props.theme.primary};
        color: ${props => props.theme.primaryColor};
        transition: all 0.35s; 
        display: inline-block;
        border-radius: 15px;
        padding: 8px 20px;
        box-shadow: 1px 8px 19px 3px rgba(61,55,61,0.25);
        user-select: none;

        &:hover {
            cursor: pointer;
            background-color: ${props => props.theme.primaryLight};
            color: ${props => props.theme.primaryLightColor}
            box-shadow: 1px 8px 19px 3px rgba(61,55,61,0.48);
        }
    }
`

function Button (props) {
    let buttonData = {
        className: 'button',
        onClick: props.onClickFunction,
        onMouseOver: props.onMouseOverFunction
    }

    return (
        <StyleWrapper>
            <button {...buttonData }>
                { props.label }
            </button>
        </StyleWrapper>
    )
}

export default Button