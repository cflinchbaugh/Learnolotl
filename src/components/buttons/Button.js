import React from 'react';
import styled from 'styled-components';

const StyleWrapper = styled.div`
    .button {
        color: ${props => props.theme.primaryColor};
        background: transparent;
        transition: all 0.35s; 
        display: inline-block;
        border: solid 1px orange;
        border-radius: 15px;
        padding: 8px 20px;
        box-shadow: 1px 8px 19px 3px rgba(61,55,61,0.25);
        user-select: none;

        &:hover {
            cursor: pointer;
            background-color: #566184;
            color: ${props => props.theme.primaryLightColor}
            box-shadow: 1px 8px 19px 3px rgba(61,55,61,0.48);
        }
    }
`

function Button (props) {
    let buttonData = {
        className: 'button',
        onClick: props.onClickFunction,
        onMouseOver: props.onMouseOverFunction
    }

    return (
        <StyleWrapper>
            <button {...buttonData }>
                { props.label }
            </button>
        </StyleWrapper>
    )
}

export default Button