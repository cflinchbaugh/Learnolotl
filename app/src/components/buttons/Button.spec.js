import React from 'react';
import {shallow} from 'enzyme';
import Button from '../Button';

describe('Button', () => {
  it('should call onClickFunction when the button is clicked', () => {
    const onClickFunction = jest.fn();

    const wrapper = shallow(<Button onClickFunction={onClickFunction}/>);

        wrapper.find('.button').simulate('click');

        expect(onClickFunction).toHaveBeenCalled();
  });
});