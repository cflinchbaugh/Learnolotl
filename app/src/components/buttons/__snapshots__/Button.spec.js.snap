// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders and matches snapshot 1`] = `
<styled.div>
  <button
    className="button"
    onClick={undefined}
    onMouseOver={undefined}
  />
</styled.div>
`;
